### PR TITLE
[BugFix] Fix not support projectNode for short circuit read

### PR DESCRIPTION
**Pull request by Jay-ju**
Why I'm doing:

What I'm doing:

Fixes #issue
fix  issue: https://github.com/StarRocks/starrocks/issues/38515

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195460592。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195460592.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195461903。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195461903.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195462997。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195462997.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195464444。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195464444.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195465483。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195465483.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195466905。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195466905.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195468081。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195468081.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195469447。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195469447.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195521124。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195521124.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195522474。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195522474.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195523646。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195523646.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195524498。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195524498.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195525701。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195525701.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195526954。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195526954.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195527953。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195527953.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195528993。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195528993.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Review by Seaven — APPROVED** [review]

**Review by decster — APPROVED** [review]

File: be/src/exec/short_circuit.cpp [hunk 1]
```diff
@@ -187,17 +187,15 @@ Status ShortCircuitExecutor::execute() {
     ChunkPtr chunk;
     bool eos = false;
     while (true) {
-        RETURN_IF_ERROR(_source->get_next(runtime_state(), &chunk, &eos));
```
Comment (satanson) [inline be/src/exec/short_circuit.cpp:190]: non-pipeline engine is deprecated and discarded,  we havn't to implement non-pipeline execution.

**Pull request merged**